### PR TITLE
Rename metadata to properties

### PR DESCRIPTION
**Pull request by roznawsk**
# Rename `Component.Metadata` to `Component.Properties`

Motivation:
The name "Metadata" interpreted as Data about Data is inaccurate in this case, since the Metadata structure contains information about component, such as `playable?`, `lowLatency?` or `targetWindowDuration` in the case of HLS component.

I think that the proposed name `Properties` is much better in this case and a lot more self-explanatory.

## Acknowledging the stipulations set forth:
 > - [ ] I hereby confirm that a Pull Request involving updates to the Software Development Kit (SDK) has been smoothly merged, currently awaits processing, or is otherwise deemed unnecessary in this context.

The changes to the SDKs will follow after implementing the File Endpoint
 - [x] I also affirm that another Pull Request, specifically addressing updates to the documentation body (commonly referred to as 'docs'), has either been successfully incorporated, is in the process of review, or is considered superfluous under the prevailing circumstances.

### CODE REVIEW

File: test/test_helper.exs
Comment (Rados13) [inline test/test_helper.exs:5]: Isn't this change only for the debugging purpose?

**Review by Rados13 — COMMENTED** [review]
Based on this comment
```
The changes to the SDKs will follow after implementing the File Endpoint
```
This PR is blocked until File Endpoint is merge?

**Review by mickel8 — COMMENTED** [review]
I agree that properties > metadata but did we consider `options` , `config` and `configuration`?

**Review by mickel8 — APPROVED** [review]

**Review by Karolk99 — APPROVED** [review]

**Pull request merged**
